presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -715,6 +715,35 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
         return distances[rightCodePoints.length - 1];
     }
 
+    @Description("computes Hamming distance between two strings")
+    @ScalarFunction
+    @LiteralParameters({"x", "y"})
+    @SqlType(StandardTypes.BIGINT)
+    public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
+    {
+        int distance = 0;
+        int leftPosition = 0;
+        int rightPosition = 0;
+        while (leftPosition < left.length() && rightPosition < right.length()) {
+            int codePointLeft = tryGetCodePointAt(left, leftPosition);
+            int codePointRight = tryGetCodePointAt(right, rightPosition);
+
+            if (codePointLeft != codePointRight) {
```

Comment:
Let's add a comment before this to state: "if both code points are invalid, treat them equal". Otherwise people may argue it's a bug.